salesagility/SuiteCRM
# Add functionality allowing for file attachment of notes via the V8 API.

To add functionality allowing for file attachment of notes via the V8 API. 

## Description
This allows file attachment of notes via the API. It requires use of the filename (which should include file extension) and filecontents (which should be base64 encoded) parameters in the API request. Other than these two parameters it should work the same as any other insert or update API call.

## Motivation and Context
Currently this is not possible via the API.

## How To Test This
This can be tested by created API calls to create, update and delete a note. For the create and update calls, you should include the filename and filecontents parameters (ensure these are the right format as stated above). Ensure that the file is added/changed to the record in the CRM and also check the record is correct in the upload directory (the name in this directory should match the ID of the record).  For delete, ensure it deletes both the record and the file from the upload directory.

Should be tested with multiple file types. 

## Types of changes


### Final checklist
